// @picovoice/eagle-node/dist/index.js
//
// Copyright 2024 Picovoice Inc.
//
// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
// file accompanying this source.
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
// specific language governing permissions and limitations under the License.
//
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInt16Frames = exports.checkWaveFile = exports.EagleErrors = exports.EagleProfilerEnrollFeedback = exports.Eagle = exports.EagleProfiler = void 0;
const eagle_1 = require("./eagle");
Object.defineProperty(exports, "EagleProfiler", { enumerable: true, get: function () { return eagle_1.EagleProfiler; } });
Object.defineProperty(exports, "Eagle", { enumerable: true, get: function () { return eagle_1.Eagle; } });
const wave_util_1 = require("./wave_util");
Object.defineProperty(exports, "getInt16Frames", { enumerable: true, get: function () { return wave_util_1.getInt16Frames; } });
Object.defineProperty(exports, "checkWaveFile", { enumerable: true, get: function () { return wave_util_1.checkWaveFile; } });
const types_1 = require("./types");
Object.defineProperty(exports, "EagleProfilerEnrollFeedback", { enumerable: true, get: function () { return types_1.EagleProfilerEnrollFeedback; } });
const EagleErrors = require("./errors");
exports.EagleErrors = EagleErrors;
