// @picovoice/eagle-node/dist/types.js
"use strict";
/*
  Copyright 2024 Picovoice Inc.
  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
  file accompanying this source.
  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  specific language governing permissions and limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.EagleProfilerEnrollFeedback = void 0;
var EagleProfilerEnrollFeedback;
(function (EagleProfilerEnrollFeedback) {
    EagleProfilerEnrollFeedback["NONE"] = "NONE";
    EagleProfilerEnrollFeedback["AUDIO_TOO_SHORT"] = "AUDIO_TOO_SHORT";
    EagleProfilerEnrollFeedback["UNKNOWN_SPEAKER"] = "UNKNOWN_SPEAKER";
    EagleProfilerEnrollFeedback["NO_VOICE_FOUND"] = "NO_VOICE_FOUND";
    EagleProfilerEnrollFeedback["QUALITY_ISSUE"] = "QUALITY_ISSUE";
})(EagleProfilerEnrollFeedback = exports.EagleProfilerEnrollFeedback || (exports.EagleProfilerEnrollFeedback = {}));
